### ISSUE TITLE
info-initialize should be called before helm-get-info-files

### ISSUE THREAD

**Issue by arunisaac** [emacs-helm/helm#2118]
## Expected behavior

When `helm-info` is run, it should populate `helm-default-info-index-list` with info files from `Info-directory-list`.

## Actual behavior (from `emacs-helm.sh` if possible, see note at the bottom)

`helm-get-info-files` is called before emacs has initialized `Info-directory-list`. Hence, only info files from `Info-default-directory-list` are detected.

## Solution

Call `info-initialize` in `helm-get-info-files` before trying to read `Info-directory-list`. Currently, I am working around this bug with the following advice.

```
(advice-add 'helm-get-info-files :before 'info-initialize)
```

## Describe versions of Helm, Emacs, operating system, etc.

helm-3.0
emacs-26.1
Operating System: GNU GuixSD (rolling release)

## Are you using `emacs-helm.sh` to reproduce this bug (yes/no):

No, I did not try `emacs-helm.sh`, but I was able to reproduce this bug using `emacs -Q`. Therefore, there could not be any interference from other packages.

Thanks.

**Comment by thierryvolpiatto** [comment]
Arun Isaac <notifications@github.com> writes:

> Expected behavior
>
> When helm-info is run, it should populate helm-default-info-index-list with info files from Info-directory-list.
>
> Actual behavior (from emacs-helm.sh if possible, see note at the bottom)
>
> helm-get-info-files is called before emacs has initialized
> Info-directory-list. Hence, only info files from
> Info-default-directory-list are detected.
>
> Solution
>
> Call info-initialize in helm-get-info-files before trying to read
> Info-directory-list.

This could help only if you have set env var INFOPATH.

--
Thierry

**Comment by arunisaac** [comment]
> This could help only if you have set env var INFOPATH.

Yes, I have the INFOPATH environment variable set, and it contains a
bunch of non-standard locations not included in
`Info-default-directory-list`.

**Comment by thierryvolpiatto** [comment]
Arun Isaac <notifications@github.com> writes:

>> This could help only if you have set env var INFOPATH.
>
> Yes, I have the INFOPATH environment variable set, and it contains a
> bunch of non-standard locations not included in
> `Info-default-directory-list`.

Same here, and starting from emacs-helm.sh all is loaded properly, did I
miss something?

> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub, or mute the thread.*


-- 
Thierry

**Comment by thierryvolpiatto** [comment]
Ah! `emacs-helm.sh` is calling `package-initialize` which calls indirectly `info-initialize`, perhaps you are not calling `package-initialize`?
I will make the change you request anyway to fix this use case.

**Comment by arunisaac** [comment]
> Ah! `emacs-helm.sh` is calling `package-initialize` which calls
> indirectly `info-initialize`, perhaps you are not calling
> `package-initialize`?

Yes, I am not calling `package-initialize`. I only use my distribution's
(GuixSD) package manager to manage emacs packages. I don't use
package.el.

> I will make the change you request anyway to fix this use case.

Thanks! :-)
